### ISSUE TITLE
Implement `watch` config option for Jest automatic configuration.

### ISSUE THREAD

**Issue by Guardiannw** [wallabyjs/public#2415]
### Issue description or question

Currently, there seems to be one way to run the `globalSetup` and `globalTeardown` code only once in watch mode with Jest, namely, to check for the `watch` or `watchAll` config options (see  [https://github.com/facebook/jest/issues/6800](https://github.com/facebook/jest/issues/6800)).  However, wallaby doesn't override this setting, although it seems that it should when it is going to be run continuously.  This would enable the ability to run complex `globalSetup` and `globalTeardown` code only once throughout the "watched" execution.

**Comment by smcenlly** [comment]
Thanks for providing the jest issue. This was useful in us understanding what's going on from a jest perspective when people only want to run `globalSetup` and `globalTeardown` once in watch mode. What they're doing in jest is actually a bit of a hack (albeit, it works), but it relies on how Jest is being run in watch mode and the fact that the jest process stays alive between executions. Unfortunately in Wallaby, this is not the case, we sandbox and restart a new Jest run every time your tests are executed or change so the same approach isn't supported.

For Wallaby, what you're going to need to do is to perform your global setup logic that you want to "stay alive" between test executions in a wallaby configuration file. You can have a mix of both automatic configuration and a wallaby setup file to achieve this.

The example below shows how to do this:

```javascript
const process = require('process');

// Declare your module export with 2 parameters.
// The second parameter is your "done" function that returns your wallaby configuration
module.exports = async (wallaby, done) => {
  await new Promise(resolve => {
    setTimeout(() => {
      process.env.DATABASEURL = 'my database url';
      resolve()
    }, 2500);
  });

  done({
      autoDetect: true
  });
};
```

You would need to refactor your jest tests to allow them to run side-by-side with Wallaby. For example, you could set a process.env variable to let jest setup/teardown functions know that Wallaby is running and to do nothing.

In terms of teardown, as mentioned in other issues, teardown is never guaranteed so your setup needs to consider that. If you need a teardown, you may try something like [`signal-exit`](https://www.npmjs.com/package/signal-exit) that was mentioned in the jest issue, but we're not 100% sure that this will work or that it will play well with Wallaby.

**Comment by smcenlly** [comment]
We have updated Wallaby to pass an an environment variable to our test runner processes, `WALLABY_ENV` with a value of `true`. In addition to being able to create a Wallaby.js setup file as we previously mentioned, you may now instead switch on this process.env variable in your jest setup/teardown code. We've provided a sample below:

**Jest Global Setup**
```javascript
module.exports = async () => {
  // return if Wallaby and setup is already done
  if (process.env.WALLABY_ENV && process.env.SETUP_DONE)
    return;

  process.env.SETUP_DONE = true;

  // setup code
};
```

**Jest Global Teardown**
```javascript
module.exports = async () => {
  // return if Wallaby
  if (process.env.WALLABY_ENV)
    return;

  // teardown code
};
```

As per our previous advice, if you need some more aggressive cleanup code, you may need to try something like `signal-exit` for the process but there is still no guarantee that this will be run.
